frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/unload-arriving-vehicles/simulated-region-overview-behavior-unload-arriving-vehicles.component.ts

```diff
@@ -8,4 +23,93 @@ import { Component } from '@angular/core';
         './simulated-region-overview-behavior-unload-arriving-vehicles.component.scss',
     ],
 })
-export class SimulatedRegionOverviewBehaviorUnloadArrivingVehiclesComponent {}
+export class SimulatedRegionOverviewBehaviorUnloadArrivingVehiclesComponent
+    implements OnInit
+{
+    @Input()
+    simulatedRegionId!: UUID;
+
+    @Input()
+    behaviorId!: UUID;
+
+    unloadDuration$?: Observable<number>;
+    vehiclesStatus$?: Observable<VehicleUnloadStatus[]>;
+
+    constructor(
+        private readonly exerciseService: ExerciseService,
+        public readonly store: Store<AppState>
+    ) {}
+
+    ngOnInit(): void {
+        const simulatedRegionSelector = createSelectSimulatedRegion(
+            this.simulatedRegionId
+        );
+        const behaviorStateSelector = createSelector(
+            simulatedRegionSelector,
+            (simulatedRegion) =>
+                simulatedRegion?.behaviors.find(
+                    (behavior) => behavior.id === this.behaviorId
+                ) as UnloadArrivingVehiclesBehaviorState | undefined
+        );
+        this.unloadDuration$ = this.store.select(
+            createSelector(
+                behaviorStateSelector,
+                (behavior) => behavior?.unloadDelay ?? 0
+            )
+        );
+
+        const activityStateSelector = createSelector(
+            simulatedRegionSelector,
+            (simulatedRegion) =>
+                Object.values(simulatedRegion.activities).filter(
+                    (activity): activity is UnloadVehicleActivityState =>
+                        activity.type === 'unloadVehicleActivity'
+                )
+        );
+
+        const unloadingSelector = createSelector(
+            activityStateSelector,
+            selectVehicles,
+            (activities, vehicles) =>
+                activities.map((activity) => ({
+                    vehicle: vehicles[activity.vehicleId]!,
+                    endTime: activity.startTime + activity.duration,
+                }))
+        );
+
+        const currentTimeSelector = createSelector(
+            selectExerciseState,
+            (state) => state.currentTime
+        );
```

Comment:
There already is `selectCurrentTime` in `frontend/src/app/state/application/selectors/exercise.selectors.ts`